telemetry/packages/public-app/components/preLoader.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import '../app/globals.css';
+import { useEffect, useState } from 'react';
+import GridDisplay from './gridDisplay';
+import Image from 'next/image';
+
+const style: React.CSSProperties = {
+  backgroundColor: 'black',
+  color: 'white',
+  fontSize: '20px',
+  justifyContent: 'center',
+  alignContent: 'center',
+  alignItems: 'center', // corrected property name from alignContent to alignItems
+  display: 'flex',
+  flexDirection: 'column', // added display property to center content horizontally
+  height: '70vh', // added height to center content vertically
+  width: '100vw',
+  margin: 'auto',
+};
+
+export default function PreLoader(): JSX.Element {
+  //const [data, setData] = useState([])
+  const [loading, setLoading] = useState<boolean>(true);
+  const [processing, setProcessing] = useState<boolean>(true);
+  const [processed, setProcessed] = useState<boolean>(true);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setProcessing(false);
+    }, 1500);
+    setTimeout(() => {
+      setProcessed(false);
+    }, 3000);
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setLoading(false);
+      document.documentElement.style.setProperty(
+        '--bg-color',
+        document.documentElement.style.getPropertyValue('--bg-color') ===
+          'black'
+          ? 'white'
+          : 'black',
+      );
```

Comment:
This will need to be changed because we can't use browser-specific APIs like `document` in Next.js.